tests/unit/test_vtk.py

```diff
@@ -1,9112 +1,409 @@
-1.00000000000e+00
-1.00000000000e+00
-
-</DataArray>
-<DataArray type="Float64" Name="dummy_vector" NumberOfComponents="3" format="ascii">
-1.00000000000e+00
-1.00000000000e+00
-1.00000000000e+00
-1.00000000000e+00
-1.00000000000e+00
-1.00000000000e+00
-1.00000000000e+00
-1.00000000000e+00
-1.00000000000e+00
-
-</DataArray>
-</CellData>
-</Piece>
-</UnstructuredGrid>
-</VTKFile>
-"""
-
-    def _test_fractures_3d_vtu(self):
-        return """<?xml version="1.0"?>
-<VTKFile type="UnstructuredGrid" version="0.1" byte_order="LittleEndian">
-<!--This file was created by -->
-<UnstructuredGrid>
-<Piece NumberOfPoints="8" NumberOfCells="2">
-<Points>
-<DataArray type="Float64" Name="Points" NumberOfComponents="3" format="ascii">
-0.00000000000e+00
-0.00000000000e+00
-0.00000000000e+00
-1.00000000000e+00
-0.00000000000e+00
-0.00000000000e+00
-2.00000000000e+00
-1.00000000000e+00
-1.00000000000e+00
-0.00000000000e+00
-1.00000000000e+00
-1.00000000000e+00
-5.00000000000e-01
--1.00000000000e+00
-2.00000000000e+00
-5.00000000000e-01
--1.00000000000e+00
--1.00000000000e+00
-5.00000000000e-01
-2.00000000000e+00
--1.00000000000e+00
-5.00000000000e-01
-2.00000000000e+00
-2.00000000000e+00
-
-</DataArray>
-</Points>
-<Cells>
-<DataArray type="Int64" Name="connectivity" format="ascii">
-0
-1
-2
-3
-4
-5
-6
-7
-
-</DataArray>
-<DataArray type="Int64" Name="offsets" format="ascii">
-4
-8
-
-</DataArray>
-<DataArray type="Int64" Name="types" format="ascii">
-7
-7
-
-</DataArray>
-</Cells>
-<CellData>
-<DataArray type="Int64" Name="dummy_scalar" format="ascii">
-1
-1
-
-</DataArray>
-<DataArray type="Float64" Name="dummy_vector" NumberOfComponents="3" format="ascii">
-1.00000000000e+00
-1.00000000000e+00
-1.00000000000e+00
-1.00000000000e+00
-1.00000000000e+00
-1.00000000000e+00
-
-</DataArray>
-<DataArray type="Int64" Name="fracture_number" format="ascii">
-1
-2
-
-</DataArray>
-</CellData>
-</Piece>
-</UnstructuredGrid>
-</VTKFile>
-"""
-
-    def _cross_platform_integer_type(self, txt: str) -> str:
-        """Ensures that the primitive integer type is cross-platform
-        Int types in NumPy are platform-dependent, see:
-        https://github.com/numpy/numpy/issues/9464
-        """
-        # Default integer type in windows is Int32
-        if sys.platform == "win32":
-            return txt.replace("Int64", "Int32")
-        else:
-            return txt
-
-
-if __name__ == "__main__":
-    unittest.main()
+        # Define ingredients of where to store vtu files for exporting during testing.
+        self.folder = "./test_vtk/"
+        self.file_name = "grid"
+        self.folder_reference = folder_reference
+
+
+@pytest.fixture
+def setup():
+    # Method to deliver a setup to all tests, and remove any temporary directory
+
+    # Setup
+    setup = ExporterTestSetup()
+    yield setup
+
+    # Teardown
+    shutil.rmtree(
+        os.getcwd() + "/" + os.path.basename(os.path.normpath(setup.folder)),
+        ignore_errors=True,
+    )
```

Comment:
The point is that with Path objects / is interpreted independent of the operating system.

Reply:
On linux this works (modulo some simple syntax changes). @IvarStefansson could you check whether this raises trouble on Windows?

Reply:
I'm pretty sure it will not work because windows wants backslash instead of slash. That is the point of pathlib.Path, it takes care of OS-specific cases, and will for instance translate the / (note: not a string) to the right separator.

Reply:
My mistake. "this" meant the proposed changes by @keileg and the original. See line 52.